### ISSUE TITLE
Future annotations cause add_arguments "NameError" when using Serializable

### ISSUE THREAD

**Issue by zhiruiluo** [lebrice/SimpleParsing#181]
**Describe the bug**
the ```from __future__ import annotations``` cause add_arguments "NameError" when using Serializable
It will raise the following error:
"NameError: name 'D' is not defined"

**To Reproduce**
```python
# test_simple_parsing.py
from __future__ import annotations
from simple_parsing import Serializable, ArgumentParser, subgroups
from dataclasses import dataclass
import pytest

@dataclass
class MyArguments(Serializable):
    arg1: str = 'this_argment'
    
@pytest.mark.parametrize(
    'sys_argv, result', [
        (['test.py'], 'this_argment'),
        (['test.py', '--arg1', 'test2'], 'test2')
    ],
)
def test_simple_parsing(sys_argv, result):
    parser = ArgumentParser()
    parser.add_arguments(MyArguments, 'myargs')
    args, _ = parser.parse_known_args(sys_argv)
    assert args.myargs.arg1 == result
```

**Expected behavior**

```console
$ pytest test_simple_parsing.py
2 passed
```

**Actual behavior**
A clear and concise description of what is happening.

```console
$ pytest test_simple_parsing.py
2 failed
NameError: name 'D' is not defined
```

**Desktop (please complete the following information):**
 - simple_parsing version: 0.0.21
 - Python version: 3.9.13

**Additional context**

**Comment by zhiruiluo** [comment]
Absuolutely, we will pass the test by removing the '''from __future__ import annotations''' from this file.

**Comment by zhiruiluo** [comment]
Here is my proposed fix #182.
Added an entry to the forward_refs_to_types in get_field_annotation.py to fix the error.
Added a test_issue_181.py file about this issue.

**Comment by zhiruiluo** [comment]
Looks like that this is related to #168.

**Comment by zhiruiluo** [comment]
Thanks for review my code, @lebrice.
The fix is enclosed in #183.
I am closing this issue now.
